### DISCUSSION TITLE
Decimal to do list

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#370] [General]
1. new register->fields:
  register->declength    // allocated/available storage length
  register->digits           // number of assigned digits 
2. way of converting floats to decimals. they are needed for float-to-decimal instructions
   *** Who can provide me with a decent printf format to use for floats?
4. <s>We don't have integer-to-decimal arithmetic yet. I think they're needed as well.       </s>        *** integrated with  last pushed version 
5. Do we need an integer-to-decimal comparison? As we will have register-to-float comparisons, I guess so.  
6. There was no real testing, mainly copy/paste. 
5.1 we need to test if the instructions is called at all
5.2 does it deliver a meaningful result
5.3 there are no range limitations or status testings of the DEC functions in place. This must be done!

**Comment by rvjansen** [comment]
3) We could always do this via Strings. I wonder how efficiency is (we are not going for efficiency in variable precision decimal) and how that efficiency offsets against having more instructions in the interpreter; certainly in the non-threaded version of that.

**Reply by Peter-Jacob** [reply]
That would offend my honour if I had one. It is not a big deal, just 3 or 4 for the arithmetic. I started already ;)

**Reply by rvjansen** [reply]
ok, great

**Reply by Peter-Jacob** [reply]
it's part of the latest commit.  Decimal Integer comparison I'll implement later.

**Comment by Peter-Jacob** [comment]
I placed all items in Discussion Thread #378
